datapath: Remove !CONNTRACK (v2)

brb opened the pull request:
99% of Cilium functionality depends on CONNTRACK being enabled, so it
doesn't make sense to complicate our codebase with handling !CONNTRACK
case.

This is the second attempt to remove !CONNTRACK. The diff from the
previous attempt:

* Fix the "RuntimeConntrackInVethModeTest Conntrack-related
  configuration options for endpoints" test case.
* Mention the removal in the upgrade guide.
* Deprecate `--disable-conntrack`.

brb committed:
datapath: Remove !CONNTRACK

99% of Cilium functionality depends on CONNTRACK being enabled, so it
doesn't make sense to complicate our codebase with handling !CONNTRACK
case.

This is the second attempt to remove !CONNTRACK. The diff from the
previous attempt:

* Fix the "RuntimeConntrackInVethModeTest Conntrack-related
  configuration options for endpoints" test case.
* Mention the removal in the upgrade guide.

Signed-off-by: Martynas Pumputis <m@lambda.lt>

qmonnet reviewed (commented):
Is there any point in keeping the `--disable-conntrack` (and the `option.Conntrack`) for the agent, or should they go as well? It looks it's only used to override the endpoint options (no longer necessary after your PR) when we want a L4 policy?

nathanjsweet reviewed (approved):
Deleting datapath code is hawt. We can deprecate the conntrack option now, no?

brb committed:
daemon: Deprecate and make no-op --disable-conntrack

Considering the previous commit, it is no longer makes sense to keep the
option. Previously, the option was not even exposed via Helm.

Suggested-by: Quentin Monnet <quentin@isovalent.com>
Suggested-by: Nate Sweet <nathanjsweet@pm.me>
Signed-off-by: Martynas Pumputis <m@lambda.lt>

qmonnet commented on Documentation/operations/upgrade.rst:
```diff
@@ -319,6 +319,15 @@ New Options
+  to disable the stateful connection tracking for the endpoint. However, many
+  Cilium features depend on the tracking. Therefore the option to disable
+  the connection tracking was removed. In addition, we deprecated
+  the ``disable-conntrack`` option and made it non-operational.
```
Maybe mention that we intend to remove it in v1.13?

```suggestion
  the connection tracking was removed.
* Related to the above, option ``disable-conntrack`` for the agent has been
  deprecated and made non-operational. It will be removed in version 1.13.
```

qmonnet reviewed (approved):
Looks good (one minor suggestion below), thanks!

Do you know if there is any use case for disabling conntrack statistics at runtime, or if we can also let go of the `ConntrackAccounting` option in the future?
